plugins/processors/flux/flux.go

```diff
@@ -0,0 +1,342 @@
+package flux
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/csv"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/lang"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/telegraf/plugins/processors/flux/tables"
+	// Cannot import builtin because we need to register telegraf.from().
+	_ "github.com/influxdata/flux/stdlib"
+	"github.com/influxdata/flux/values"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+const (
+	description  = "Runs a flux script to process inputs and produce outputs"
+	sampleConfig = `
+  ## Path to a Flux script.
+  path = "path/to/script.flux"
+  ## Enables debug mode if an output file is specified.
+  debug = "path/to/out.csv"
+`
+	measurementColumnLabel = "_measurement"
+)
+
+type Flux struct {
+	Path  string `toml:"path"`
+	Debug string `toml:"debug"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	debugFile *os.File
+	program   flux.Program
+}
+
+func init() {
+	processors.Add("flux", func() telegraf.Processor {
+		return &Flux{}
+	})
+}
+
+func (p *Flux) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Flux) Description() string {
+	return description
+}
+
+func (p *Flux) init() error {
```

Comment:
Please make this
```suggestion
func (p *Flux) Init() error {
```
which is called automatically by telegraf after parsing and initializing the config options.